src/main/java/com/suse/salt/netapi/calls/LocalCall.java

```diff
@@ -178,13 +164,85 @@ public LocalCall(String functionName, Optional<List<?>> arg,
             final SaltClient client, Target<?> target,
             String username, String password, AuthModule authModule)
             throws SaltException {
+        return callSyncBatchable(client, target, Optional.of(username),
+                Optional.of(password), Optional.of(authModule), Optional.empty()).get(0);
+    }
+
+    /**
+     * Calls a execution module function on the given target with batching and
+     * synchronously waits for the result. Authentication is done with the token
+     * therefore you have to login prior to using this function.
+     *
+     * @param client SaltClient instance
+     * @param target the target for the function
+     * @param batch  the batch parameter, e.g. 10, 10%, etc.
+     * @return A list of maps with each list representing each batch, and maps containing
+     * the results with the minion names as keys.
+     * @throws SaltException if anything goes wrong
+     */
+    public List<Map<String, Result<R>>> callSyncBatch(
+            final SaltClient client, Target<?> target, String batch)
```

Comment:
Even if salt represents batch as a string i think it a little bit too broad and error prone. It would be nice to have it be a real type + some helpers to construct them like `Batch.percent(20)` or `Batch.amount(200)`.